Core/Categories/Category.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace Core.Categories
+{
+    public abstract class Category
+    {
+        public Category(string name, double increment, double capacity)
+        {
+            Name = name;
+            Increment = increment;
+            Capacity = capacity;
+        }
+
+        public string Name { get; }
+        public double Increment { get; }
+        public double Capacity { get; }
+        
+        public override bool Equals(object? obj)
+        {
+            return obj is Category category &&
+                   Name == category.Name &&
+                   Increment == category.Increment &&
+                   Capacity == category.Capacity;
```

Comment:
Codacy found an issue: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=6003477)